### ISSUE TITLE
Inefficient Training and Redundant Sampler Implementation

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#1690]
Inadequate Training Process and Sampler Implementation

The current implementation of the training process creates a new optimizer instance at each epoch, leading to inefficient training, and the Sampler class is redundant as it can be achieved through DataLoader with shuffle=True.
